### DISCUSSION TITLE
Почему Filter.removeTracks не удаляет треки?

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#100] [Помощь с алгоритмом] [answered]
Из массива `sourceArray` не удаляются треки `removedArray`. Что я делаю не так?
```js
let sourceArray = Source.getTracks([
  { name: 'ALL', id: '1Xdpab5CROu8A57f1Y1ly9' },
]);

let removedArray = Source.getSavedTracks([
  { name: 'One-Two', id: '27yfUjdakxxb5pftd1qNCO' },
]);

Filter.removeTracks(sourceArray, removedArray);

Playlist.saveAsNew({
  name: 'deletingTest',
  public: false,
  tracks: sourceArray,
});
```

**Comment by Chimildic** [comment, marked as answer]
В `removedArray` запрашиваете любимые треки `getSavedTracks`, а не треки плейлиста. Передаваемый аргумент с id игнорируется, поскольку эта функция для другого. Замените на `getTracks`. Если плейлист только один, можно `getPlaylistTracks`. Описание всех функций в [документации](https://chimildic.github.io/goofy/#/func?id=source).

Для тестирования лучше использовать [отладку](https://chimildic.github.io/goofy/#/best-practices?id=Выполнение-отладки).
```js
function test() {
  let sourceArray = Source.getPlaylistTracks('', '1Xdpab5CROu8A57f1Y1ly9');
  let removedArray = Source.getPlaylistTracks('', '27yfUjdakxxb5pftd1qNCO');

  Filter.removeTracks(sourceArray, removedArray);

  Playlist.saveAsNew({
    name: 'deletingTest',
    public: false,
    tracks: sourceArray,
  });
}
```

**Reply by Bondian** [reply]
Спасибо! Конечно же! Мои извинения за невнимательность и поспешность... (тупо скопировал с какого-то примера, не разувая глаз). Постараюсь быть внимательней
